core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -1446,12 +1456,23 @@ object Parser {
     *  this is useful for parsing comment strings, for instance.
     */
   def until0(p: Parser0[Any]): Parser0[String] =
-    (not(p).with1 ~ anyChar).rep0.string
+    (not(p).void.with1 ~ anyChar).rep0.string
 
   /** parse one or more characters as long as they don't match p
     */
   def until(p: Parser0[Any]): Parser[String] =
-    (not(p).with1 ~ anyChar).rep.string
+    (not(p).void.with1 ~ anyChar).rep.string
+
+  /** parse zero or more times until Parser `end` succeeds.
+    */
+  def until0[A](p: Parser[A], end: Parser[Any]): Parser0[List[A]] = {
+    (not(end).void.with1 ~ p).rep0.map(_.map(_._2))
```

Comment:
Can we use *> so we don't need the map? We can also remove the void on the left since *> will do that.